converter/auto/auto_examples_test.go

```diff
@@ -0,0 +1,34 @@
+package auto
+
+import (
+	"fmt"
+	"github.com/grafana/agent/component/loki/source/kafka"
+	"github.com/grafana/loki/clients/pkg/promtail/scrapeconfig"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestKafka(t *testing.T) {
```

Comment:
This test reveals that in this approach we have plenty of exceptions to add,  making this approach probably more error-prone than manually spelling out the entire conversion.